dracut/30afterburn/afterburn-hostname.service

```diff
@@ -0,0 +1,24 @@
+[Unit]
+Description=Afterburn Hostname
+# Azure needs to fetch the hostname in the initramfs
+ConditionKernelCommandLine=|ignition.platform.id=azure
+
+# This will be firstboot only to set up the correct
+# hostname for kube.
+ConditionFirstBoot=yes
+
+After=initrd-root-fs.target
+Wants=initrd-root-fs.target
+
+# We order this service after disks have been provisioned
+# but before ignition-files stage has run, so ignition can
+# overwrite the configured hostname with e.g. a static one
+After=ignition-disks.service
+Before=ignition-files.service
+
```

Comment:
This is maybe more personal preference, but I like to cut down the `After`/`Before` directives to what is actually necessary to make it clearer what the dependencies really are. So e.g. here, we should *only* need `After=initrd-root-fs.target` and `Before=ignition-files.service`. (We don't need `After=ignition-disks.service`, since clearly that must happen after `/sysroot` is mounted, and we don't need `Wants=initrd-root-fs.target` since that's already part of the transaction.)

That said, if you've already validated this works, I'm cool rolling with it like that as well!

Reply:
I agree, I just quickly added those in. Let me clean it up a bit, thanks for the suggestion!

Reply:
On that note, do you know where we're using https://github.com/coreos/ignition-dracut/blob/spec2x/dracut/30ignition/ignition-remount-sysroot.service ? Is that something we should look out for

Reply:
We don't need it in RHCOS and FCOS, see https://github.com/coreos/ignition-dracut/commit/3ec0b39106a0724982da64dc348a38da7a471467. It was added back for other distros, see https://github.com/coreos/ignition-dracut/issues/37 and https://github.com/coreos/ignition-dracut/pull/38#issuecomment-451965981.